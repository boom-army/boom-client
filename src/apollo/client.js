import { ApolloClient, InMemoryCache } from "@apollo/client";
import { offsetLimitPagination } from "@apollo/client/utilities";

export const client = new ApolloClient({
  cache: new InMemoryCache({
    typePolicies: {
      Query: {
        fields: {
          feed: offsetLimitPagination(),
          users: offsetLimitPagination(),
        },
      },
    },
  }),
  uri: process.env.REACT_APP_APOLLO_API || "http://locahost:7777",
  headers: {
    authorization: localStorage.getItem("token") || "",
  },
  context: async ({ req }) => {
    const token = req?.headers?.authorization || null;
    const user= token ? JSON.parse(localStorage.getItem("user")) : null;
    return { user, isLoggedIn: !!token };
  },
});
