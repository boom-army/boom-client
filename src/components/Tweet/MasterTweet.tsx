import boomLogo from "../../images/logo.png";
import { styled } from "@mui/material/styles";
import { CustomResponse } from "../CustomResponse";
import { Helmet } from "react-helmet";
import { Loader } from "../Loader";
import { NewTweet, ParentTweet, ShowTweet } from ".";
import { useMeQuery, useTweetQuery } from "../../generated/graphql";
import { useParams } from "react-router-dom";
import { Box } from "@mui/material";

export const MasterTweet = () => {
  const { tweetId } = useParams();

  const { data, loading } = useTweetQuery({
    variables: {
      id: tweetId!,
    },
  });
  const { data: userData } = useMeQuery();

  const comments =
    data?.tweet?.childTweets?.length! > 0 ? data?.tweet.childTweets : [];
  const exists = !!data?.tweet?.id;
  const hasParent = !!data?.tweet?.parentTweet?.id;

  let heroImage = boomLogo;
  if (data?.tweet?.files?.length) heroImage = data.tweet.files[0].url;
  if (data?.tweet?.nft?.image) heroImage = data.tweet.nft.image;

  return (
    <Box mb={7}>
      {/* <Helmet>
        <title>Boom</title>
        <meta
          name="title"
          content={`Meep on app.boom.army by ${data?.tweet?.user?.handle}`}
        />
        <meta name="description" content={data?.tweet?.text} />

        <meta name="og:url" content={window.location.href} />
        <meta
          name="og:title"
          content={`Meep on app.boom.army by ${data?.tweet?.user?.handle}`}
        />
        <meta name="og:description" content={data?.tweet?.text} />
        <meta name="og:image" content={heroImage} />

        <meta name="twitter:url" content={window.location.href} />
        <meta
          name="twitter:title"
          content={`Meep on app.boom.army by ${data?.tweet?.user?.handle}`}
        />
        <meta name="twitter:description" content={data?.tweet?.text} />
        <meta name="twitter:image" content={heroImage} />
      </Helmet> */}
      {loading ? (
        <Loader />
      ) : (
        <>
          {hasParent && (
            <ParentTweet parentTweet={data?.tweet?.parentTweet?.id} />
          )}
          {exists ? (
            <ShowTweet tweet={data.tweet} />
          ) : (
            <CustomResponse text="Oops, the tweet you are looking for doesn't seem to exist." />
          )}
          {exists && userData && <NewTweet parentTweet={data?.tweet?.id} userData={userData?.me} />}
          {comments &&
            comments.map((comment: any) => (
              <ShowTweet tweet={comment} key={comment.id} />
            ))}
        </>
      )}
    </Box>
  );
};
