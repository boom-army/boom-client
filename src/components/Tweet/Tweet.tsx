import React from "react";
import "linkify-plugin-hashtag";
import "linkify-plugin-mention";
import Linkify from "linkify-react";
import UserAvatar from "../UserAvatar";
import moment from "moment";
import { CommentIcon } from "../Icons";
import { EmojiTweet, Retweet } from "./index";
import { ImageBox } from "../ImageBox";
import { LAMPORTS_PER_SOL } from "../../constants/math";
import { Link } from "react-router-dom";
import { List as ReactionsList } from "../Reactions/List";
import { NFTTweet } from "../NFT/NFTTweet";
import { TipCreator } from "../TipCreator";
import { styled } from "@mui/material/styles";
import { Tweet } from "../../generated/graphql";
import { VideoContainer } from "../Giphy/VideoContainer";
import { setDate } from "../../utils";
import { useReaction } from "../../hooks/useReaction";

const Wrapper = styled("div")((props) => ({
  display: "flex",
  borderBottom: `1px solid ${props.theme.tertiaryColor}`,
  padding: "1.5rem 1rem 1rem 1rem",

  ".avatar": {
    marginRight: "1em",
    marginBottom: "1em",
  },

  ".tweet-info": {
    width: "100%",
  },

  ".tweet-info-user": {
    display: "flex",
  },

  ".tweet-info-user span.username": {
    fontWeight: 500,
  },

  ".tweet-info-user span.secondary": {
    paddingLeft: "0.5rem",
    color: props.theme.secondaryColor,
  },

  ".tweet-body": {
    marginBottom: "0.75rem",
    wordBreak: "break-word",
  },

  ".tags": {
    display: "flex",
  },

  "a.body": {
    color: props.theme.accentColor,
  },

  "div.tweet-stats": {
    display: "inline-flex",
    flexWrap: "wrap",
    alignItems: "center",

    div: {
      color: props.theme.secondaryColor,
    },

    ".controls": {
      marginRight: "4rem",
    },

    svg: {
      marginRight: "0.5rem",
    },

    span: {
      display: "flex",
      alignItems: "center",
    },

    "span.comment": {
      svg: {
        position: "relative",
        top: "4px",
      },
    },
  },

  "@media screen and (max-width: 470px)": {
    "div.tweet-stats": {
      div: {
        marginRight: "1.5rem",
      },
      ".controls": {
        marginRight: "1.5rem",
      },
    },
  },

  "@media screen and (max-width: 430px)": {
    flexDirection: "column",

    ".username": {
      display: "none",
    },

    ".avatar": {
      display: "none",
    },

    ".tweet-info-user span.secondary": {
      paddingLeft: "0",
      paddingRight: "0.5rem",
    },
  },
}));

interface Props {
  tweet: Tweet;
}

export const ShowTweet: React.FC<Props> = ({ tweet }:Props) => {
  const {
    id,
    text,
    // tags,
    user,
    files,
    gif,
    nft,
    // isTweetMine,
    isRetweet,
    retweetsCount,
    reactions,
    commentsCount,
    tipsCount,
    createdAt,
  } = tweet;

  const { handleReaction } = useReaction({ tweetId: id });
  const handle = user && user.handle;

  const linkifyOptions = {
    className: "body",
    target: { url: "_blank" },
    formatHref: { hashtag: (href: any) => `explore?=${href.substring(1)}` },
  };
  return (
    <Wrapper>
      <Link to={`/${handle}`}>
        <UserAvatar className="avatar" avatar={user?.avatar as string} />
      </Link>

      <div className="tweet-info">
        <div className="tweet-info-user">
          <Link to={`/${handle}`}>
            <span className="username">{user && user.consumerName}</span>
            <span className="secondary">{`@${handle}`}</span>
          </Link>
          &nbsp;&nbsp;
          <Link to={`/${handle}/status/${id}`} className="secondary">
            {moment(setDate(createdAt)).fromNow()}
          </Link>
        </div>

        <Linkify options={linkifyOptions}>
          <p className="tweet-body">{text}</p>
        </Linkify>

        {gif && <VideoContainer gif={gif} />}

        {nft && <NFTTweet nftData={nft} />}

        {!!files.length && <ImageBox files={files} disablelightbox={false} />}

        {reactions.length > 0 && (
          <ReactionsList
            reactions={reactions}
            handleReaction={handleReaction}
            tweetId={id}
          />
        )}

        <div className="tweet-stats">
          <EmojiTweet handleReaction={handleReaction} />

          <div className="controls">
            <span className="comment">
              <Link to={`/${handle}/status/${id}`}>
                <CommentIcon />
                {commentsCount ? commentsCount : null}
              </Link>
            </span>
          </div>

          <div className="controls">
            <Retweet
              id={id}
              isRetweet={isRetweet}
              retweetsCount={retweetsCount}
            />
          </div>

          <>
            <TipCreator
              userPubKey={user?.publicAddress}
              tipAmount={tipsCount && parseInt(tipsCount) / LAMPORTS_PER_SOL}
              tweetId={id}
              userId={user?.id}
            />
          </>

          {/* <div>
            <span>{isTweetMine ? <DeleteTweet id={id} /> : null}</span>
          </div> */}
        </div>
      </div>
    </Wrapper>
  );
};
