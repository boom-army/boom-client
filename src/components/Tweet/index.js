export { DeleteTweet } from './DeleteTweet';
export { EmojiTweet } from './EmojiTweet';
export { MasterTweet } from './MasterTweet';
export { NewTweet } from './NewTweet';
export { ParentTweet } from './ParentTweet';
export { Retweet } from './Retweet';
export { ShowTweet } from './Tweet';
