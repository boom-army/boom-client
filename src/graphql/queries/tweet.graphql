query tweet($id: ID!) {
  tweet(id: $id) {
    ...TweetData
    masterTweets {
      ...TweetData
    }
  }
}

mutation newTweet(
  $text: String!
  $files: [String!]!
  $tags: [String!]!
  $mentions: [String!]!
  $gif: GifInput
  $nft: NFTInput
  $parentTweet: String
  $masterTweet: String
  $channel: String
) {
  newTweet(
    text: $text
    files: $files
    tags: $tags
    mentions: $mentions
    gif: $gif
    nft: $nft
    parentTweet: $parentTweet
    masterTweet: $masterTweet
    channel: $channel
  ) {
    id
    text
    tags
    mentions
    commentsCount
    createdAt
  }
}

mutation toggleRetweet($id: ID!) {
  toggleRetweet(id: $id)
}

mutation deleteTweet($id: ID!) {
  deleteTweet(id: $id) {
    id
  }
}
