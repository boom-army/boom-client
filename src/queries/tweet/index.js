import gql from "graphql-tag";

export const NEW_TWEET = gql`
  mutation newTweet(
    $text: String!
    $files: [String!]!
    $tags: [String!]!
    $mentions: [String!]!
    $gif: GifInput
    $nft: NFTInput
    $parentTweet: String
  ) {
    newTweet(
      text: $text
      files: $files
      tags: $tags
      mentions: $mentions
      gif: $gif
      nft: $nft
      parentTweet: $parentTweet
    ) {
      id
      text
      tags
      mentions
      commentsCount
      createdAt
    }
  }
`;

export const DELETE_TWEET = gql`
  mutation deleteTweet($id: ID!) {
    deleteTweet(id: $id) {
      id
    }
  }
`;

export const TOGGLE_RETWEET = gql`
  mutation toggleRetweet($id: ID!) {
    toggleRetweet(id: $id)
  }
`;

export const TOGGLE_REACTION = gql`
  mutation toggleReaction($id: ID!, $emojiId: String!, $skin: Int) {
    toggleReaction(id: $id, emojiId: $emojiId, skin: $skin)
  }
`;

const TWEET_FRAGMENT = gql`
  fragment subTweet on Tweet {
    id
    text
    tags
    mentions
    retweetsCount
    isRetweet
    createdAt
    parentTweet {
      id
    }
    user {
      id
      publicAddress
      consumerName
      handle
      avatar
    }
    files {
      id
      url
    }
    gif {
      id
      title
      fixedHeightUrl
      originalUrl
    }
    nft {
      id
      publicKey
      name
      symbol
      description
      sellerFeeBasisPoints
      externalUrl
      image
      attributes {
        traitType
        value
      }
      collection {
        name
        family
      }
      properties {
        files {
          uri
          type
        }
        category
        creators {
          address
          share
        }
      }
    }
    reactions {
      id
      emojiId
      skin
      isMine
      count
    }
  }
`;

export const TWEET = gql`
  query tweet($id: ID!) {
    tweet(id: $id) {
      ...subTweet
      childTweets {
        ...subTweet
        childTweets {
          ...subTweet
        }
      }
    }
  }
  ${TWEET_FRAGMENT}
`;
