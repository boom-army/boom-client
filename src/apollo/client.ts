import { ApolloClient, InMemoryCache } from "@apollo/client";
import { offsetLimitPagination } from "@apollo/client/utilities";

export const client = new ApolloClient({
  cache: new InMemoryCache({
    typePolicies: {
      Query: {
        fields: {
          feed: offsetLimitPagination(),
          heroFeed: offsetLimitPagination(),
          getChannelById: offsetLimitPagination(),
          users: offsetLimitPagination(),
          searchTweets: offsetLimitPagination(),
          searchUser: offsetLimitPagination(),
          mentions: offsetLimitPagination(),
        },
      },
    },
  }),
  credentials: 'include',
  uri: process.env.REACT_APP_APOLLO_API || "http://locahost:7777",
  headers: {
    authorization: localStorage.getItem("token") || "",
    "Access-Control-Allow-Origin" : "*",
  },
  defaultOptions: {
    watchQuery: {
      context: async ({ req }: any) => {
        const token = req?.headers?.authorization || null;
        const user = token ? localStorage.getItem("user")?.toString() : null;
        return { user, isLoggedIn: !!token };
      },
    },
  },
});
