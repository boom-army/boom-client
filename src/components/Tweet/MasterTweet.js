import React from "react";
import boomLogo from "../../images/logo.png"
import styled from "styled-components";
import useMetaTags from "react-metatags-hook";
import { CustomResponse } from "../CustomResponse";
import { Loader } from "../Loader";
import { META_METAS, META_LINKS, META_OG } from "../../constants/meta";
import { NewTweet, ParentTweet, ShowTweet } from ".";
import { TWEET } from "../../queries/tweet";
import { useParams } from "react-router-dom";
import { useQuery } from "@apollo/client";

const Wrapper = styled.div`
  margin-bottom: 7rem;
`;

export const MasterTweet = () => {
  const { tweetId } = useParams();

  const { loading, data } = useQuery(TWEET, {
    variables: { id: tweetId },
  });

  const comments =
    data?.tweet?.childTweets?.length > 0 ? data.tweet.childTweets : [];
  const exists = !!data?.tweet?.id;
  const hasParent = !!data?.tweet?.parentTweet?.id;

  let heroImage = boomLogo;
  if (data?.tweet?.files?.length) heroImage = data.tweet.files[0].url;
  if (data?.tweet?.nft?.image) heroImage = data.tweet.nft.image;

  useMetaTags(
    {
      metas: [...META_METAS],
      links: [ ...META_LINKS ],
      openGraph: META_OG,
      twitter: {
        title: `Meep on app.boom.army by ${data?.tweet?.user?.handle}`,
        description: data?.tweet?.text,
        image: heroImage,
      },
    },
    [data]
  );
  

  return (
    <Wrapper>
      {loading ? (
        <Loader />
      ) : (
        <>
          {hasParent && (
            <ParentTweet parentTweet={data?.tweet?.parentTweet?.id} />
          )}
          {exists ? (
            <ShowTweet
              tweet={data && data.tweet}
            />
          ) : (
            <CustomResponse text="Oops, the tweet you are looking for doesn't seem to exist." />
          )}
          {exists && <NewTweet parentTweet={data?.tweet?.id} />}
          {comments &&
            comments.map((comment) => (
              <ShowTweet
                tweet={comment && comment}
                key={comment.id}
              />
            ))}
        </>
      )}
    </Wrapper>
  );
};
