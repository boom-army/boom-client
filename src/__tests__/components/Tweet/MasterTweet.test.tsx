import React from "react";
import { MasterTweet } from "../../../components/Tweet/MasterTweet";
import renderer from "react-test-renderer";
import { MockedProvider } from "@apollo/client/testing";
// import { MentionsDocument, Tweet } from "../../../generated/graphql";
import {TOGGLE_REACTION, TWEET } from "../../../queries/tweet/index";
// import { TOGGLE_REACTION, TWEET } from "../queries/tweet";
import { act, fireEvent, render } from "@testing-library/react"; 
import { MemoryRouter } from "react-router-dom";
import { SnackbarProvider } from "notistack"

export const __mocks__:any= [
    {
      request: {
       query: TWEET,
        variables: {
             id:"cl084xdy815761s0n5omoxoem0" 
        },
      
      },
      result: {
        data: {
          mentions:[]
        ,
        },
        errors: "An error occurred",
      },
  }
  ]

const Tweet: any = {
  id: "cl084xdy815761s0n5omoxoem0",
  text: "testing after changes in input converted to ts",
  tags: [],
  isTweetMine: true,
  commentsCount: 1,
  retweetsCount: 0,
  isRetweet: false,
  tipsCount: "0",
  createdAt: "1646139451616",
  parentTweet: null,
  files: [],
  gif: null,
  nft: null,
  user: {
    id: "ckzny1iv10031lmn568gj3b3n",
    publicAddress: "JBZ52cKhHiFJdzQBNWnp3Xy2jHgDgkhWwSLWoNgLNxD4",
    avatar: "",
    handle: "shy-cloud-4965",
    consumerName: "shy-cloud-4965",
    __typename: "User",
  },
  reactions: [
    {
      id: "cl0c774p371312s0n533u0b29k",
      emojiId: "+1",
      skin: 1,
      isMine: true,
      count: 1,
      __typename: "Reaction",
    },
  ],
  __typename: "Tweet",
};



test("Master Tweet Snapshot testing ...", async () => {
  let rendered;
  await act(async () => {
    rendered = render(
      <MockedProvider mocks={__mocks__} addTypename={false}>
        <SnackbarProvider>
       <MasterTweet  />
       </SnackbarProvider>
      </MockedProvider>,{wrapper: MemoryRouter}
    );
  });
  expect(rendered).toMatchSnapshot();
});
