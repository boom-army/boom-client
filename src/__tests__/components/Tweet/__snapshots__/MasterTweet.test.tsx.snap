// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Master Tweet Snapshot testing ... 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": @keyframes animation-0 {
  0% {
    -webkit-transform: rotate(0deg);
    -moz-transform: rotate(0deg);
    -ms-transform: rotate(0deg);
    transform: rotate(0deg);
  }

  100% {
    -webkit-transform: rotate(360deg);
    -moz-transform: rotate(360deg);
    -ms-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}

@keyframes animation-1 {
  0% {
    stroke-dasharray: 1px,200px;
    stroke-dashoffset: 0;
  }

  50% {
    stroke-dasharray: 100px,200px;
    stroke-dashoffset: -15px;
  }

  100% {
    stroke-dasharray: 100px,200px;
    stroke-dashoffset: -125px;
  }
}

.css-0 {
  margin-bottom: 7rem;
}

.css-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.css-2 {
  display: inline-block;
  color: #1976d2;
  -webkit-animation: animation-0 1.4s linear infinite;
  animation: animation-0 1.4s linear infinite;
}

.css-3 {
  display: block;
}

.css-4 {
  stroke: currentColor;
  stroke-dasharray: 80px,200px;
  stroke-dashoffset: 0;
  -webkit-animation: animation-1 1.4s ease-in-out infinite;
  animation: animation-1 1.4s ease-in-out infinite;
}

<body>
    <div>
      <div
        class="css-0"
      >
        <div
          class="MuiBox-root css-1"
        >
          <span
            class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-2"
            role="progressbar"
            style="width: 30px; height: 30px;"
          >
            <svg
              class="MuiCircularProgress-svg css-3"
              viewBox="22 22 44 44"
            >
              <circle
                class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-4"
                cx="44"
                cy="44"
                fill="none"
                r="20.2"
                stroke-width="3.6"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": @keyframes animation-0 {
  0% {
    -webkit-transform: rotate(0deg);
    -moz-transform: rotate(0deg);
    -ms-transform: rotate(0deg);
    transform: rotate(0deg);
  }

  100% {
    -webkit-transform: rotate(360deg);
    -moz-transform: rotate(360deg);
    -ms-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}

@keyframes animation-1 {
  0% {
    stroke-dasharray: 1px,200px;
    stroke-dashoffset: 0;
  }

  50% {
    stroke-dasharray: 100px,200px;
    stroke-dashoffset: -15px;
  }

  100% {
    stroke-dasharray: 100px,200px;
    stroke-dashoffset: -125px;
  }
}

.css-0 {
  margin-bottom: 7rem;
}

.css-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.css-2 {
  display: inline-block;
  color: #1976d2;
  -webkit-animation: animation-0 1.4s linear infinite;
  animation: animation-0 1.4s linear infinite;
}

.css-3 {
  display: block;
}

.css-4 {
  stroke: currentColor;
  stroke-dasharray: 80px,200px;
  stroke-dashoffset: 0;
  -webkit-animation: animation-1 1.4s ease-in-out infinite;
  animation: animation-1 1.4s ease-in-out infinite;
}

<div>
    <div
      class="css-0"
    >
      <div
        class="MuiBox-root css-1"
      >
        <span
          class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-2"
          role="progressbar"
          style="width: 30px; height: 30px;"
        >
          <svg
            class="MuiCircularProgress-svg css-3"
            viewBox="22 22 44 44"
          >
            <circle
              class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-4"
              cx="44"
              cy="44"
              fill="none"
              r="20.2"
              stroke-width="3.6"
            />
          </svg>
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
